js/msp.js

```diff
@@ -1507,7 +1507,7 @@ MSP.crunch = function (code) {
             break;
             
         case MSP_codes.MSP_SET_NAME:
-            for (var i = 0; i<32; i++) {
+            for (var i = 0; i<CONFIG.name.length; i++) {
```

Comment:
I would probably change the end condition to something like `i<CONFIG.name.length && i < 32` here, to avoid causing an error on the firmware side in case something wrongly sets `CONFIG.name` to a string that is longer than what the firmware expects. (e.g., once this can be backed up / restored, somebody manually editing the config file...)